health/module.go

```diff
@@ -0,0 +1,55 @@
+package health
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/textileio/powergate/net"
+)
+
+// Module exposes the filecoin wallet api.
+type Module struct {
+	net net.Module
+}
+
+// Status represents the node's health status
+type Status int
+
+const (
+	// Ok specifies the node is healthy
+	Ok Status = iota
+	// Degraded specifies there are problems with the node health
+	Degraded
+	// Error specifies there was an error when determining node health
+	Error
+)
+
+// New creates a new net module
+func New(net net.Module) *Module {
+	m := &Module{
+		net: net,
+	}
+	return m
+}
+
+// Check reutuns the current health status and any messages related to the status
+func (m *Module) Check(ctx context.Context) (status Status, messages []string, err error) {
```

Comment:
Just makes sure we have an acceptable connection status to each of our peers. This can be improved in the future